tasks/rh-sign-cosign-image/rh-sign-cosign-image.yaml

```diff
@@ -0,0 +1,103 @@
+              key: AWS_ACCESS_KEY_ID
+        - name: AWS_SECRET_ACCESS_KEY
+          valueFrom:
+            secretKeyRef:
+              name: $(params.secretName)
+              key: AWS_SECRET_ACCESS_KEY
+      script: |
+        #!/usr/bin/env bash
+        set -eu
+        function fetch_cosign_key() {
+          echo $(kubectl_get_arg_from_config_map $1 SIG_KEY)
+        } 
+
+        SNAPSHOT_PATH=$(workspaces.data.path)/$(params.snapshotPath)
+
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No valid data file was provided."
+            exit 1
+        fi
+
+        request=$(jq -r '.sign.request // "hacbs-signing-pipeline"' ${DATA_FILE})
+        CONFIG_MAP_NAME=$(jq -rc '.sign.configMapName // "signing-config-map"' ${DATA_FILE})
+
+        COMPONENTS_LENGTH=$(jq '.components |length' ${SNAPSHOT_PATH})
+
+        COSIGN_SIG_KEY=$(fetch_cosign_key $CONFIG_MAP_NAME)
+
+        for (( COMPONENTS_INDEX=0; COMPONENTS_INDEX<COMPONENTS_LENGTH; COMPONENTS_INDEX++ )); do
+            INTERNAL_CONTAINER_IMAGE=$(jq -r ".components[${COMPONENTS_INDEX}].containerImage" ${SNAPSHOT_PATH})
+            INTERNAL_CONTAINER_REF="${INTERNAL_CONTAINER_IMAGE/@*}"
+            DIGEST="${INTERNAL_CONTAINER_IMAGE/*@}"
+
+            PUB_CONTAINER_REF=$(jq -r ".components[${COMPONENTS_INDEX}].repository" ${SNAPSHOT_PATH})
+            PUB_CONTAINER_IMAGE="${PUB_CONTAINER_REF}@${DIGEST}"
+
+            COSIGN_PRE_ARGS="AWS_REGION=${AWS_REGION}"
+            COSIGN_PRE_ARGS="$COSIGN_PRE_ARGS AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID}"
+            COSIGN_PRE_ARGS="$COSIGN_PRE_ARGS AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}"
+            COSIGN_COMMON_ARGS="--key ${COSIGN_SIG_KEY}"
+
+            IMAGE=$(skopeo inspect --raw "docker://${INTERNAL_CONTAINER_IMAGE}")
+            MEDIA_TYPE=$(echo "$IMAGE" | jq -r '.mediaType')
+            LIST=0
+            if [ "$MEDIA_TYPE" = "application/vnd.docker.distribution.manifest.list.v2+json" ]; then LIST=1; fi
+            if [ "$MEDIA_TYPE" = "application/vnd.oci.image.index.v1+json" ]; then LIST=1; fi
+
+            # if manifest list, sign each manifest + ml itself
+            if [ $LIST -eq 1 ]; then
+              PUB_REPO_REF="${PUB_CONTAINER_REF}@${DIGEST}"
+              for MDIGEST in $(echo "$IMAGE" | jq -rc '.manifests[]|.digest'); do
+                cosign ${COSIGN_PRE_ARGS} cosign -t 3m0s sign\
+                  ${COSIGN_COMMON_ARGS}\
+                  --sign-container-identity "${PUB_CONTAINER_REF}@${MDIGEST}"\
+                  "${INTERNAL_CONTAINER_REF}@${MDIGEST}"
+              done
+            fi
+              # if sign ml itself or just manifest (if it's not list)
+              cosign ${COSIGN_PRE_ARGS} cosign -t 3m0s sign\
+                ${COSIGN_COMMON_ARGS}\
+                --sign-container-identity "${PUB_CONTAINER_REF}@${DIGEST}"\
+                "${INTERNAL_CONTAINER_REF}@${DIGEST}"
```

Comment:
The indentation here should be 2 spaces less.

Reply:
fixed

Reply:
I still see wrong indentation:
```sh
            fi
              # if sign ml itself or just manifest (if it's not list)
              echo "Signing manifest ${INTERNAL_CONTAINER_REF}@${DIGEST}"
              cosign -t 3m0s sign\
              ${COSIGN_COMMON_ARGS}\
              --sign-container-identity "${PUB_CONTAINER_REF}@${DIGEST}"\
              "${INTERNAL_CONTAINER_REF}@${DIGEST}"
            echo "done"
```
Also, the comment "if sign..." - probably the "if" should be dropped, because now there's no if:
So should be:
```sh
            fi
            # sign ml itself or just manifest (if it's not list)
            echo "Signing manifest ${INTERNAL_CONTAINER_REF}@${DIGEST}"
            cosign -t 3m0s sign\
              ${COSIGN_COMMON_ARGS}\
              --sign-container-identity "${PUB_CONTAINER_REF}@${DIGEST}"\
              "${INTERNAL_CONTAINER_REF}@${DIGEST}"
            echo "done"
```

Reply:
Fixed in https://github.com/konflux-ci/release-service-catalog/pull/467.

Reply:
Thanks.